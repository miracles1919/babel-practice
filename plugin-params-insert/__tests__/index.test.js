const pluginTester = require('babel-plugin-tester').default;

const InsertParamsPlugin = require('../plugin/index');

const sourceCode = `
  console.log('hello world')

  function fun() {
    console.log('foo')
  }

  export default class App {

    say() {
      console.log('what');
    }

    render() {
      return (
        <>
          <div onClick={() => console.log('click')}>app</div>
          <div>{console.log('jsx')}</div>
        </>
        
      )
    }
  }
`;

pluginTester({
  plugin: InsertParamsPlugin,
  pluginName: 'InsertParamsPlugin',
  babelOptions: {
    parserOpts: {
      sourceType: 'unambiguous',
      plugins: ['jsx'],
    },
  },
  tests: {
    'console.log': {
      code: sourceCode,
      snapshot: true,
    },
  },
});
