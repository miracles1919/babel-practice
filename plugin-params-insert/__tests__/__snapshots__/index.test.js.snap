// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsertParamsPlugin console.log: console.log 1`] = `

console.log('hello world')

function fun() {
  console.log('foo')
}

export default class App {

  say() {
    console.log('what');
  }

  render() {
    return (
      <>
        <div onClick={() => console.log('click')}>app</div>
        <div>{console.log('jsx')}</div>
      </>
      
    )
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

console.log("line: 1, column: 0");
console.log("hello world");

function fun() {
  console.log("line: 4, column: 2");
  console.log("foo");
}

export default class App {
  say() {
    console.log("line: 10, column: 4");
    console.log("what");
  }

  render() {
    return (
      <>
        <div
          onClick={() => {
            console.log("line: 16, column: 28");
            console.log("click");
          }}
        >
          app
        </div>
        <div>{[console.log("line: 17, column: 14"), console.log("jsx")]}</div>
      </>
    );
  }
}


`;
